quinn-h3/src/client.rs

```diff
@@ -488,8 +488,8 @@ impl Connection {
     ///
     /// All ongoing requests will be completed, all new requests or server push will
     /// be rejected.
-    pub fn go_away(&mut self) {
-        self.0.h3.lock().unwrap().inner.go_away();
+    pub fn go_away(&mut self, grace: u64) {
```

Comment:
What is the `grace` value here? Maybe add a comment about that? Alternatively, I think you've named it `max_id` in other places, which is a bit easier to understand.

Reply:
I changed this from max_id because now it is the number of potentially in-flight streams to gracefully accept before starting rejecting. Here it could be called `allow_push`, and in `servers.rs` it could be `allowed_request`. Do you think these names would be a better fit?

Reply:
Yeah, those sound good!